dot/digest/digest.go

```diff
@@ -249,159 +188,13 @@ func (h *Handler) handleBlockFinalisation(ctx context.Context) {
 				h.logger.Errorf("failed to persist babe next epoch config: %s", err)
 			}
 
-			err = h.handleGrandpaChangesOnFinalization(info.Header.Number)
+			err = h.grandpaState.ApplyScheduledChanges(&info.Header)
 			if err != nil {
-				h.logger.Errorf("failed to handle grandpa changes on block finalisation: %s", err)
+				h.logger.Errorf("failed to apply standard scheduled changes on block finalisation: %w", err)
```

Comment:
```suggestion
				h.logger.Errorf("failed to apply scheduled change on block finalisation: %w", err)
```

not sure what's meant by standard :p also, there should only be one scheduled change per block right? or at least, only one that can take effect

Reply:
substrate uses `standard` and `forced` scheduled changes, removed the `standard` term, and yes just one scheduled change should be applied in the `addScheduledChange` method we have a check to avoid adding more than one scheduled change per block. However, there is an edge case where in the same block there are scheduled and forced changes, in that case, we should apply only the forced change.

Reply:
okay, makes sense!